Auth.php

```diff
@@ -547,15 +547,22 @@ public function checkSession($hash)
         $currentdate = strtotime(date("Y-m-d H:i:s"));
         $db_ip = $row['ip'];
         $db_cookie = $row['cookie_crc'];
+        $db_device_id = $row['device_id'];
 
         if ($currentdate > $expiredate) {
             $this->deleteSession($hash);
 
             return false;
         }
 
-        if ($ip != $db_ip) {
-            return false;
+        if($device_id != null){
+            if($db_device_id !== $device_id){
+                return false;
+            }
+        } else {
```

Comment:
What is a device id? Where cames the id from? Comparing the php session_id is more usefull.

Reply:
i got those from the ionic framework: https://github.com/Paldom/UniqueDeviceID which my app is based on. Every device has his own id. with it i can make sure that the user uses the same device.

Reply:
What I see at he ionic framework is: 
```protected void getDeviceId(){
        try {
            Context context = cordova.getActivity().getApplicationContext();
            TelephonyManager tm = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);

            String uuid;
            String androidID = Secure.getString(context.getContentResolver(), Secure.ANDROID_ID);
            String deviceID = tm.getDeviceId();
            String simID = tm.getSimSerialNumber();

            if ("9774d56d682e549c".equals(androidID) || androidID == null) {
                androidID = "";
            }

            if (deviceID == null) {
                deviceID = "";
            }

            if (simID == null) {
                simID = "";
            }

            uuid = androidID + deviceID + simID;
            uuid = String.format("%32s", uuid).replace(' ', '0');
            uuid = uuid.substring(0, 32);
            uuid = uuid.replaceAll("(\\w{8})(\\w{4})(\\w{4})(\\w{4})(\\w{12})", "$1-$2-$3-$4-$5");

            this.callbackContext.success(uuid);
        }catch(Exception e ) {
            this.callbackContext.error("Exception occurred: ".concat(e.getMessage()));
        }```

So this uniquedeviceid isn't something provided by the hardware, but something derived per case, since they  have IOS, and So son... 
 
This isn't like a generalized  use case but a specific usage with that specific library... maybe if you generalize the solution to db_custom_id !=  custom_id ... and the user can implement ?

Reply:
sure i can rename it to db_custom_id... but which datatype in db should it have? at the moment it is varchar(36) for v4 uuid.

Reply:
varchar 200, is  a good size and still indexeable... make it '' by default